week-4/Homework/mandatory/3-water-bottle.js

```diff
@@ -18,23 +18,44 @@ You have to implement the missing features according to the specification.
 */
 
 // Here is your starting point:
+// calling this function should pour your bottle full (volume = 100);
 let bottle = {
   volume: 0,
-  fillUp: function() {
-    // calling this function should pour your bottle full (volume = 100);
+  fillUp: function () {
+    this.volume += 100;
   },
+  // calling this function should increase your bottle volume by 10 unit;
   pour: function () {
-    // calling this function should increase your bottle volume by 10 unit;
+    if (this.volume <= 90) {
+      this.volume += 10;
+    }
   },
-  drink: function() {
-    // calling this function should decrease your bottle volume by 10 unit;
+
+  // calling this function should decrease your bottle volume by 10 unit;
+
+  drink: function () {
+    if (this.volume >= 10) {
+      this.volume = this.volume - 10;
+    }
   },
+
+  // this function should return true if your bottle is full;
   isFull: function () {
-    // this function should return true if your bottle is empty;
+    if (this.volume === 100) {
+      return true;
+    } else {
+      return false;
+    }
+  },
+
+  // this function should return true if your bottle is empty;
+  isEmpty: function () {
+    if (this.volume === 0) {
+      return true;
+    } else {
+      return false;
+    }
   },
-  isEmpty: function() {
-    // this function should return true if your bottle is full;
-  }
 };
 
 /*

```

Comment:
there is a missing part which you have been asked to answer and here it's 'What do you think why it is preferred using `this` inside object over its variable name in our case `bottle`?'

Reply:
done!